.github/workflows/nightly-upstream-snapshot-build.yml

```diff
@@ -94,19 +102,319 @@ jobs:
           build-args: "ADOT_JAVA_VERSION=${{ env.ADOT_JAVA_VERSION }}"
           context: .
           platforms: linux/amd64,linux/arm64
-          tags: |
-            ${{ env.IMAGE_TAG }}
-
-      - name: Set image name to output
-        id: imageNameOutput
-        run: echo "imageName=${{ env.IMAGE_NAME }}" >> "$GITHUB_OUTPUT"
+          tags: ${{ steps.imageNameOutput.outputs.rcImage}}
 
       - name: Upload to GitHub Actions
         uses: actions/upload-artifact@v3
         with:
           name: aws-opentelemetry-agent.jar
           path: otelagent/build/libs/aws-opentelemetry-agent-*.jar
 
+      - name: Build and push Sample-Apps without Auto-Instrumentation Agent
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: jibBuildWithoutAgent
+        env:
+          COMMIT_HASH: ${{ steps.generate_time_stamp.outputs.nowTimeTag }}
+
+  create-test-ref:
+    runs-on: ubuntu-latest
+    outputs:
+      testRef: ${{ steps.setRef.outputs.ref }}
+    steps:
+      - name: Set testRef output
+        id: setRef
+        run: echo "ref=terraform" >> $GITHUB_OUTPUT
+
+  # job to build testbatches for e2e integration test
+  get-testing-suites:
+    runs-on: ubuntu-latest
+    needs: [ build, create-test-ref ]
+    outputs:
+      test-case-batch-key: ${{ steps.set-batches.outputs.batch-keys }}
+      test-case-batch-value: ${{ steps.set-batches.outputs.batch-values }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: ${{ needs.create-test-ref.outputs.testRef }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          path: aws-otel-java-instrumentation
```

Comment:
nit: If checking out more than once, it should be clear in the name which repo is being checked out.

Reply:
Thanks. Updated.